napari/_vispy/canvas.py

```diff
@@ -394,6 +394,8 @@ def _process_mouse_event(
 
         # Update the cursor position
         self.viewer.cursor._view_direction = event.view_direction
+        self.viewer.cursor.button = event.button
+        self.viewer.cursor.scroll = any(event.delta)
```

Comment:
Based on local tests, most of the benchmark degragation comes from these two lines as `self.viewer.cursor` is evented model.

```
| Change   | Before [93140b93]    | After [26e09294]    |   Ratio | Benchmark (Parameter)                                                   |
|----------|----------------------|---------------------|---------|-------------------------------------------------------------------------|
| +        | 14.4±0.5μs           | 24.6±0.8μs          |    1.71 | benchmark_qt_viewer_labels.QtViewerSingleLabelsSuite.time_on_mouse_move |
| +        | 358±60ms             | 591±60ms            |    1.65 | benchmark_qt_viewer_image.QtViewerViewImageSuite.time_view_image(4096)  |
| +        | 312±20μs             | 484±50μs            |    1.55 | benchmark_qt_viewer_image.QtViewerSingleImageSuite.time_set_view_slice  |
```

Reply:
what is event.delta? Is it really only related to scroll? It's a strange name for it... (I do have vague memories of it now. I would just have expected it under (e.g.) event.scroll.delta)